#2 define car prototype

ZsoltToth opened the pull request:
Please be so kind to review my code.

ZsoltToth committed:
cars - CarPrototype.java was added and Car.java was modified to use prototype.
CarBodyStyle.java and CarProducer.java enumeration was introduced.

ZsoltToth committed:
cars - Tests for Car and CarPrototype classes were added.
 Developments has to be finished due to fire alarm :'(

ZsoltToth committed:
cars - Issue #2 define car prototype is solved.
CarManagerService.java has been extended with a new record method.
CarPrototypeManagerService.java has been added.

judit0310 commented on cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/Car.java:
```diff
@@ -47,13 +98,5 @@ public boolean equals(Object obj) {
-                ", color=" + color +
-                '}';
-    }
+
```
Missing toString

ZsoltToth replied:
done

judit0310 commented on cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java:
```diff
@@ -0,0 +1,113 @@
+package hu.uni.miskolc.iit.webdevelopment.examples.cars.model;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
```
Why import these?

ZsoltToth replied:
unused import have been removed

judit0310 commented on cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java:
```diff
@@ -0,0 +1,10 @@
+ * List of car manufacturers.
+ */
+public enum CarProducer {
+    SUZUKI,OPEL,AUDI,FERRARI, TESLA, CITROEN, FIAT
```
Missing producers

ZsoltToth replied:
as you wish

judit0310 commented on cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarPrototype.java:
```diff
@@ -0,0 +1,113 @@
+
+
+
+    public CarPrototype(CarProducer producer, String model, CarBodyStyle bodyStyle, int yearOfIntroduction, String note, int crubWeight, int grossVehicleWeight, int doorNumber, double enginePerformance) {
```
negative door number for example?

ZsoltToth replied:
data type of door number has been changed to enum

judit0310 commented on cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/service/CarPrototypeManagerService.java:
```diff
@@ -0,0 +1,50 @@
+    Collection<CarPrototype> listCarPrototypes(CarBodyStyle bodyStyle);
+
+    /**
+     * Recrod a new Car Prototype into the system.
```
*Record

judit0310 reviewed (changes requested):
Please be so kind to ...

ZsoltToth committed:
cars - code review request partially fulfilled.
Car has been extended with toString method
Unused imports from CarPrototype.java has been removed.
CarProducer.java has been extended with a few Manufacturer.

ZsoltToth committed:
cars - code review request is fulfilled.
DoorNumber.java enumeration has been added.
Javadoc has been reviewed

ZsoltToth committed:
cars - Tests have been fixed

judit0310 commented on cars/core/src/main/java/hu/uni/miskolc/iit/webdevelopment/examples/cars/model/CarProducer.java:
```diff
@@ -2,9 +2,684 @@
+ * <p>
  * List of car manufacturers.
  */
 public enum CarProducer {
```
:1st_place_medal:
